realm/realm-library/src/main/cpp/io_realm_internal_Table.cpp

```diff
@@ -712,15 +712,12 @@ JNIEXPORT void JNICALL Java_io_realm_internal_Table_nativeSetByteArray(
     if (!TBL_AND_INDEX_AND_TYPE_VALID(env, TBL(nativeTablePtr), columnIndex, rowIndex, type_Binary))
         return;
     try {
-        if (dataArray == NULL) {
-            if (!TBL_AND_COL_NULLABLE(env, TBL(nativeTablePtr), columnIndex)) {
+        if (dataArray == NULL && !TBL_AND_COL_NULLABLE(env, TBL(nativeTablePtr), columnIndex)) {
```

Comment:
old performance thing: We already verified TBL a couple of lines above. We are now doing it again in TBL_AND_COL_NULLABLE. Maybe just add a FIXME for now.